runtime/src/bank.rs

```diff
@@ -3721,16 +3791,21 @@ impl Bank {
                 };
 
                 if store {
-                    if let Some(log_messages) = transaction_log_messages.get(i).cloned().flatten() {
+                    if let TransactionExecutionResult::Executed(TransactionExecutionDetails {
+                        status,
+                        log_messages: Some(log_messages),
+                        ..
+                    }) = execution_result
+                    {
                         let mut transaction_log_collector =
                             self.transaction_log_collector.write().unwrap();
                         let transaction_log_index = transaction_log_collector.logs.len();
 
                         transaction_log_collector.logs.push(TransactionLogInfo {
                             signature: *tx.signature(),
-                            result: r.clone(),
+                            result: status.clone(),
                             is_vote,
-                            log_messages,
+                            log_messages: log_messages.clone(),
```

Comment:
Newly introduced `clone()` which might be expensive.

Reply:
This clone already existed:
```rust
if let Some(log_messages) = transaction_log_messages.get(i).cloned().flatten() {
```
https://github.com/solana-labs/solana/pull/22215/files/de19eee48e82d09bd733f6b0f74f633e7a87935f#diff-ed47b4a0198313377e091bb3957bbbc63d937805426d1b2b6de39d0a50d32a0cL3724

Reply:
It's also unavoidable unfortunately

Reply:
I see, then leave it as is.